cli/cli/commands/service/inspect/inspect.go

```diff
@@ -97,30 +119,41 @@ func run(
 
 	serviceIdentifier, err := args.GetNonGreedyArg(serviceIdentifierArgKey)
 	if err != nil {
-		return stacktrace.Propagate(err, "Expected a value for non-greedy enclave identifier arg '%v' but none was found; this is a bug in the Kurtosis CLI!", enclaveIdentifierArgKey)
+		return stacktrace.Propagate(err, "Expected a value for non-greedy enclave identifier arg '%v' but none was found; this is a bug in the Kurtosis CLI!", serviceIdentifier)
 	}
 
 	showFullUuid, err := flags.GetBool(fullUuidFlagKey)
 	if err != nil {
 		return stacktrace.Propagate(err, "Expected a value for the '%v' flag but failed to get it", fullUuidFlagKey)
 	}
 
+	outputFormat, err := flags.GetString(outputFormatKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "Expected a value for the '%v' flag but failed to get it", outputFormatKey)
+	}
+
+	outputFormat = strings.ToLower(strings.TrimSpace(outputFormat))
+	if outputFormat != "" && outputFormat != jsonOutputFormat && outputFormat != yamlOutputFormat {
+		return stacktrace.NewError("Invalid output format '%s'; must be '%v' or '%v'", outputFormat, jsonOutputFormat, yamlOutputFormat)
+	}
+
 	kurtosisCtx, err := kurtosis_context.NewKurtosisContextFromLocalEngine()
 	if err != nil {
 		return stacktrace.Propagate(err, "An error occurred creating Kurtosis Context from local engine")
 	}
 
-	if err = PrintServiceInspect(ctx, kurtosisBackend, kurtosisCtx, enclaveIdentifier, serviceIdentifier, showFullUuid); err != nil {
+	if _, err = PrintServiceInspect(ctx, kurtosisBackend, kurtosisCtx, enclaveIdentifier, serviceIdentifier, showFullUuid, outputFormat, true); err != nil {
 		// this is already wrapped up
 		return err
 	}
 	return nil
 }
 
-func PrintServiceInspect(ctx context.Context, kurtosisBackend backend_interface.KurtosisBackend, kurtosisCtx *kurtosis_context.KurtosisContext, enclaveIdentifier string, serviceIdentifier string, showFullUuid bool) error {
+func PrintServiceInspect(ctx context.Context, kurtosisBackend backend_interface.KurtosisBackend, kurtosisCtx *kurtosis_context.KurtosisContext, enclaveIdentifier string, serviceIdentifier string, showFullUuid bool, outputFormat string, shouldPrintOutputFormat bool) (map[string]interface{}, error) {
+	jsonMap := map[string]interface{}{}
```

Comment:
Why don't you define a proper struct instead? Would be easier to see what exactly is included, and also the linter would scream at you if you forget to populate a field if an extra one ever gets added.

Reply:
agreed and done, sepearate out the printing of service info and the retrieving of it.